tests/manage/test_pvresize_with_node_down.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import pytest
+from time import sleep
+import threading
+
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import tier1, ManageTest
+from ocs_ci.ocs.resources.pvc import PVC
+from ocs_ci.ocs.resources.pod import Pod, get_ceph_tools_pod
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+OCS_BUG_ID = 'test-ocs-368'
+
+NAMESPACE = ocp.OCP(kind='namespace')
+OUR_PVC = None
+POD = None
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Finalize teardown and call setup
+    """
+    def finalizer():
+        teardown()
+    request.addfinalizer(finalizer)
+    setup()
+
+
+def setup():
+    """
+    Setting up the environment for the test
+
+    Create namespace
+    Create storageclass
+    Create pvc
+    Create pod
+    """
+    NAMESPACE.create(resource_name=OCS_BUG_ID)
+    config.ENV_DATA["my_namespace"] = OCS_BUG_ID
+
+    pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+    pvc_data['metadata']['namespace'] = config.ENV_DATA["my_namespace"]
+    pvc_data['spec']['storageClassName'] = 'rook-ceph-block'
+    pvc_data['spec']['resources']['requests']['storage'] = '100Gi'
+    pvc_name = pvc_data['metadata']['name']
+    global OUR_PVC
+    OUR_PVC = PVC(**pvc_data)
+    OUR_PVC.create()
+
+    pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+    pod_data['metadata']['namespace'] = config.ENV_DATA["my_namespace"]
+    pod_data['metadata']['name'] = f'{OCS_BUG_ID}-pod'
+    first_claim = pod_data['spec']['volumes'][0]
+    first_claim['persistentVolumeClaim']['claimName'] = pvc_name
+    first_container = pod_data['spec']['containers'][0]
+    first_volume_mount = first_container['volumeMounts'][0]
+    first_volume_mount['mountPath'] = f'/mnt/{OCS_BUG_ID}'
+
+    global POD
+    POD = Pod(**pod_data)
+    POD.create()
+
+    global CEPH_TOOL
+    CEPH_TOOL = get_ceph_tools_pod()
+
+
+def teardown():
+    """
+    Cleanup
+
+    Delete pod
+    Delete pvc
+    Delete namespace
+    Delete pv
+    """
+    pv_name = OUR_PVC.backed_pv
+    POD.delete()
+    POD.delete_temp_yaml_file()
+    OUR_PVC.delete()
+    OUR_PVC.delete_temp_yaml_file()
+    NAMESPACE.delete(resource_name=OCS_BUG_ID)
+    run_cmd(f'oc delete pv {pv_name}')
+
+
+def cmd_on_my_namespace(real_cmd):
+    """
+    Run an oc command on app pod in the test namespace
+
+    Retries if the command fails.
+
+    Raises:
+        CommandFailed if a failure limit is reached
+
+    Args:
+        real_cmd (str): oc command to be run ('get pods', for example)
+    """
+    fail_limit = 100
+    setup_cmd = f'rsh {OCS_BUG_ID}-pod {real_cmd}'
+    full_cmd = f'oc -n {config.ENV_DATA["my_namespace"]} {setup_cmd}'
+    not_run = True
+    fcount = 0
+    while not_run:
+        try:
+            run_cmd(full_cmd)
+            not_run = False
```

Comment:
This is not executed on the pod you created in the setup

Reply:
I am reworking this.